packages/ag-charts-community/src/chart/series/series.ts

```diff
@@ -464,7 +464,7 @@ export abstract class Series<
     }
 
     private visibleMaybeChanged() {
-        this.ctx.seriesStateManager.registerSeries(this);
+        this.ctx.seriesStateManager.updateSeries(this);
```

Comment:
This is the actual fix - previously `destroyed()` series were having visibility updated, leading to re-registration; we now differentiate between create vs. update cases.